src/main/java/eu/europeana/metis/sandbox/service/workflow/MediaProcessingServiceImpl.java

```diff
@@ -76,21 +82,45 @@ public RecordInfo processMedia(Record record) {
     return new RecordInfo(Record.from(record, outputRdf), recordErrors);
   }
 
-  private void processResourceEntry(Record record, EnrichedRdf rdfForEnrichment,
-      MediaExtractor extractor,
-      RdfResourceEntry entry, List<RecordError> recordErrors) {
-    try (ResourceExtractionResult extraction = extractor.performMediaExtraction(entry)) {
-      if (nonNull(extraction)) {
-        // Store thumbnails
-        storeThumbnails(record, extraction.getThumbnails(), recordErrors);
-        // Add result to RDF
+  /**
+   * Processes the resource
+   *
+   * @return True if the media extraction for the resource was successful; False otherwise.
+   */
+  private boolean processResource(RdfResourceEntry resourceToProcess, Record record,
+      EnrichedRdf rdfForEnrichment, MediaExtractor extractor, List<RecordError> recordErrors,
+      boolean hasRanMainThumbnail, boolean gotMainThumbnail){
+
+    ResourceExtractionResult extraction;
+    boolean successful = false;
+
+    try{
+
+      // Check if the process for main thumbnail has happened.
+      // If not, then get main thumbnail
+      if(!hasRanMainThumbnail) {
+        extraction = extractor.performMediaExtraction(resourceToProcess, false);
+      } else {
+        extraction = extractor.performMediaExtraction(resourceToProcess, gotMainThumbnail);
+      }
```

Comment:
You could get rid of both this if statement and the hasRanMainThumbnail parameter for this method by replacing this with just line 104. Did you have a reason for not going that way?

Reply:
Yes, I did had a reason when I first started refactoring the code. I was trying to avoid repeating the same step when there was no main thumbnail. 
You are right, it is not needed this `if` statement anymore. I will fix this